misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -521,7 +540,9 @@ public IfElseStatementNode transform(IfElseStatementNode ifElseStatementNode) {
         BlockStatementNode ifBody;
         Node elseBody = null;
         if (ifElseStatementNode.elseBody().isPresent()) {
-            ifBody = formatNode(ifElseStatementNode.ifBody(), 1, 0);
+            boolean needNL = options.ifStatementFormattingOptions().elseOnNewLine();
+            ifBody = formatNode(ifElseStatementNode.ifBody(), needNL ? 0 : 1,
+                    needNL ? 1 : 0);
```

Comment:
needNL ? 0 : 1 is repeated, extract out to a variable or better yet directly assign the integer to a variable instead of needNL?